coral/src/services/router-utils/route-utils.test.tsx

```diff
@@ -0,0 +1,323 @@
+      let routeObject: RouteObject;
+
+      beforeAll(() => {
+        isFeatureFlagActiveMock.mockReturnValue(true);
+        routeObject = createRouteBehindFeatureFlag({
+          element: TestElement,
+          featureFlag: TEST_FEATURE_FLAG,
+          path: route,
+          redirectRouteWithoutFeatureFlag: routeToRedirectWithoutFeatureFlag,
+        });
+      });
+
+      afterAll(() => {
+        jest.restoreAllMocks();
+      });
+
+      it("sets the right path in the object", () => {
+        expect(routeObject.path).toEqual(route);
+      });
+
+      it("sets the given element as element in the object", () => {
+        expect(routeObject.element).toEqual(TestElement);
+      });
+    });
+  });
+
+  describe("createPrivateRoute", () => {
+    const TestElement = <div data-testid="test-element">test element</div>;
+    const privateRoute = "/private-route" as Routes;
+
+    describe("returns a route object with redirection when permission is false", () => {
+      let routeObject: RouteObject;
+
+      beforeAll(() => {
+        mockAuthUser.mockReturnValue({
+          permissions: { approveDeclineTopics: false },
+        });
+        routeObject = createPrivateRoute({
+          path: privateRoute,
+          element: TestElement,
+          permission: "approveDeclineTopics",
+          redirectUnauthorized: Routes.DASHBOARD,
+        });
+      });
+
+      afterAll(() => {
+        jest.restoreAllMocks();
+      });
+
+      it("Does not render the element and redirects when permission is false", () => {
+        customRender(routeObject.element as ReactElement, {
+          queryClient: true,
+          memoryRouter: true,
+        });
+
+        expect(screen.queryByTestId("test-element")).not.toBeInTheDocument();
+        expect(mockUseToast).toHaveBeenCalledWith({
+          message: `Not authorized`,
+          position: "bottom-left",
+          variant: "danger",
+        });
+      });
+    });
+
+    describe("returns a route object with no redirection when permission is true", () => {
+      let routeObject: RouteObject;
+
+      beforeAll(() => {
+        mockAuthUser.mockReturnValue({
+          permissions: { approveDeclineTopics: true },
+        });
+        routeObject = createPrivateRoute({
+          path: privateRoute,
+          element: TestElement,
+          permission: "approveDeclineTopics",
+          redirectUnauthorized: Routes.DASHBOARD,
+        });
+      });
+
+      afterAll(() => {
+        jest.restoreAllMocks();
+      });
+
+      it("Renders the element when permission is true", () => {
+        customRender(routeObject.element as ReactElement, {
+          queryClient: true,
+          memoryRouter: true,
+        });
+
+        expect(screen.queryByTestId("test-element")).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("filteredRoutesForSuperAdmin", () => {
```

Comment:
This is the new tests case I added, rest was only moved with the file. 

The tests are a bit rough and superficial, I didn't want to start adding rendering etc. pp. here, because it's only for a temporary time.